api/context/src/main/java/io/opentelemetry/context/propagation/MultiTextMapPropagator.java

```diff
@@ -45,13 +45,22 @@
 
   @Override
   public <C> void inject(Context context, @Nullable C carrier, TextMapSetter<C> setter) {
+    if (context == null || setter == null) {
```

Comment:
I think letting the null bubble through to the inner propagators would be better for inject.

Reply:
why is this better?

Reply:
Mostly just a gut feeling, and principle of least surprise. Also the Javadoc in https://github.com/open-telemetry/opentelemetry-java/blob/v0.17.1/api/context/src/main/java/io/opentelemetry/context/propagation/TextMapPropagator.java#L46-L53 `TextMapPropagator.composite()` states that it "simply delegates injection and extraction", so that's what I would do.

Reply:
Javadoc can certainly be updated. My opinion is that this is the safest approach, so that propagators that are not maintained by us will also be guarded against these bad inputs if they are wrapped up in a composite propagator with other "official" propagators.

Reply:
👍 I think that makes sense. If a user wants the other behavior, they can implement their own "NullNaiveMultiPropagator".